src/app.ts

```diff
@@ -165,19 +195,25 @@ export default (
       method: 'updateSecret',
     });
 
+    // I think for now we can use the taskName Prefix and edit it when we
+    // are adding additional information to the check
+    const checkResponse = await context.octokit.checks.create({
+      owner: await getOwner(context),
+      repo: '.github',
+      name: 'peribolos-run',
+      head_sha: context.payload.after,
+      status: 'queued',
+    });
     // Trigger taskrun to apply config changes to org
-    wrapOperationWithMetrics(
-      useApi(APIS.CustomObjectsApi).createNamespacedCustomObject(
-        'tekton.dev',
-        'v1beta1',
-        getNamespace(),
-        'taskruns',
-        generateTaskPayload('peribolos-run', context)
-      ),
-      {
-        install: context.payload.installation.id,
-        method: 'schedulePushTask',
-      }
+    createTaskRun('peribolos-run', context, checkResponse.data.id.toString());
+  });
+
+  app.on('check_run.rerequested', async (context: any) => {
+    // In the future a check is needed if this is the peribolos-run check
```

Comment:
We need a guard here, that the `check_run.rerequested` is about the latest commit (HEAD of default main branch). This triggers a run that is fetching/cloning the repo on master and it doesn't make sense to re-request older runs (because org config and master HEAD has to be in sync, not any outdated commit).

In cases when this is re-requested for outdated commits, we should update the check to some invalid state like skipped or ignored and we should not proceed to actually running the peribolos run.

Reply:
Resolved in a pushed commit.